app/models/concerns/orchestration/compute.rb

```diff
@@ -84,6 +84,15 @@ def setUserData
     @host      = self
     # For some reason this renders as 'built' in spoof view but 'provision' when
     # actually used. For now, use foreman_url('built') in the template
+    if template.nil?
+      failure((_("%{image} needs user data, but %{os_link} is not associated to any provisioning template of the kind user_data. Please associate it with a suitable template or uncheck 'User data' for %{compute_resource_image_link}.") %
+      { :image => image.name,
+        :os_link => "<a target='_blank' href='#{url_for(edit_operatingsystem_path(operatingsystem))}'>#{operatingsystem.title}</a>",
```

Comment:
It's a `NoMethodError` in a model unless I go overboard and include ActionView helpers here.